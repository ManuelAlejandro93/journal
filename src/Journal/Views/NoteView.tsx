import { Button, TextField, Typography } from '@mui/material';
import { SaveOutlined, CloudUploadOutlined } from '@mui/icons-material';
import { useNoteForm } from '@/Hooks';
import { SubmitEvent } from '@/Interfaces';
import { ImageGallery } from '@/Journal';

export const NoteView = () => {
  const {
    //! listos.
    formattedDate,
    body,
    title,
    //! dañados.
    noteId,
    updatedImgUrls,
    onBodyChange,
    onTitleChange,
    isFetching,
    imageInputElementRef,
    fullUpdateSingleNote
  } = useNoteForm();

  return (
    <div className='grid grid-cols-2 gap-8 animate-fade-down'>
      <Typography sx={{ color: 'primary.main', textAlign: 'center' }}>
        {formattedDate}
      </Typography>
      <Button
        disabled={isFetching}
        variant='outlined'
        sx={{ color: 'primary.main' }}
        type='submit'
        onClick={(e) => fullUpdateSingleNote(e as unknown as SubmitEvent)}
      >
        <SaveOutlined sx={{ mr: 2 }}></SaveOutlined>
        Guardar
      </Button>
      <div className='w-full col-span-2 grid space-y-6'>
        <TextField
          sx={{ color: 'primary.main' }}
          fullWidth
          placeholder='Ingresa un nuevo título'
          type='text'
          label='titulo'
          variant='filled'
          value={title}
          onChange={(e) => onTitleChange(e, noteId as string)}
        />
        <TextField
          sx={{ color: 'primary.main', gridRow: 2 }}
          fullWidth
          placeholder='Escribe aquí tu nueva nota'
          type='text'
          label='Texto de la nota'
          variant='filled'
          multiline
          minRows={3}
          value={body}
          onChange={(e) => onBodyChange(e, noteId as string)}
        />
      </div>

      <label
        className='flex items-center justify-center col-span-2'
        htmlFor='imginput'
      >
        <input
          id='imginput'
          type='file'
          accept='image/*'
          multiple
          ref={imageInputElementRef}
          disabled={isFetching}
          style={{ display: 'none' }}
        />
        <CloudUploadOutlined
          className='mr-2'
          sx={{ color: 'primary.main', fontSize: '2rem' }}
        />
        <Typography sx={{ color: 'primary.main' }}>Subir imágenes</Typography>
      </label>

      {updatedImgUrls.length <= 0 ? (
        <Typography
          className='col-span-2'
          sx={{ color: 'primary.main' }}
        >
          Sin imágenes guardadas
        </Typography>
      ) : (
        <>
          <span className='col-span-2'>
            <ol>
              {updatedImgUrls.map((img, i) => (
                <li key={img + i}>{img}</li>
              ))}
            </ol>
          </span>

          <ImageGallery />
        </>
      )}
    </div>
  );
};
//
