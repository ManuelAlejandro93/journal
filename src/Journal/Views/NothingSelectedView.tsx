import { BorderColorOutlined } from '@mui/icons-material';
import { Grid } from '@mui/material';

export const NothingSelectedView = () => {
  return (
    <Grid
      container
      spacing={0}
      direction={'column'}
      alignItems={'center'}
      justifyContent={'center'}
      sx={{
        minHeight: '80vh',
        width: '74vw',
        backgroundColor: 'primary.main',
        padding: 4
      }}
    >
      <BorderColorOutlined sx={{ color: 'white', fontSize: 120 }} />
      <h2 className='text-white text-2xl mt-4'>Crear una nueva nota</h2>
    </Grid>
  );
};
