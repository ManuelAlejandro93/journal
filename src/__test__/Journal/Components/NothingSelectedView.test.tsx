import { render, screen } from '@testing-library/react';
import { NothingSelectedView } from '../../../Journal/Views/NothingSelectedView';

describe('Pruebas en <NothingSelectedView/>', () => {
  test('Debe ser renderizable', () => {
    screen;
    const { container } = render(<NothingSelectedView />);
    expect(container).toBeTruthy();
  });

  test('Debe ser igual al snapshot', () => {
    const { container } = render(<NothingSelectedView />);
    expect(container).toMatchSnapshot();
  });

  test('El texto "Crear una nueva nota" debe ser visible en el documento', () => {
    const {} = render(<NothingSelectedView />);
    const elementoConTexto = screen.getByText(/Crear una nueva nota/i);

    expect(elementoConTexto).toBeVisible();
  });
});
