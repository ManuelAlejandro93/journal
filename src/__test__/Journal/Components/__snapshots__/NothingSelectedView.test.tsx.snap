// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NothingSelectedView/> Debe ser igual al snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-10l6qgu-MuiGrid-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kwafl0-MuiSvgIcon-root"
      data-testid="BorderColorOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m16.81 8.94-3.75-3.75L4 14.25V18h3.75zM6 16v-.92l7.06-7.06.92.92L6.92 16zm13.71-9.96c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.2-.2-.45-.29-.71-.29-.25 0-.51.1-.7.29l-1.83 1.83 3.75 3.75zM2 20h20v4H2z"
      />
    </svg>
    <h2
      class="text-white text-2xl mt-4"
    >
      Crear una nueva nota
    </h2>
  </div>
</div>
`;
